magma/conversions.py

```diff
@@ -35,6 +35,36 @@
 __all__ += ['replace']
 __all__ += ['as_bits', 'from_bits']
 
+
+class _Concatter(Finalizable):
+    def __init__(self):
+        self._ts = []
+        self._types = []
+
+    def _consume_impl(self, value):
+        if isinstance(value, ht.BitVector):
+            return value.bits(), Bits
+        if isinstance(magma_value(value), Bit):
+            return [value], Bits
+        if isinstance(value, (bool, ht.Bit)):
+            return [Bit(value)], Bits
+        if isinstance(value, Array) and isinstance(type(value).T, DigitalMeta):
+            return value.ts, type(value).abstract_t
+        raise TypeError(
+            f"expected arguments of type of Array or its subtypes, instead got "
+            f"{value} with type {type(value)}"
+        )
+
+    def consume(self, value):
+        new_ts, t = self._consume_impl(value)
+        self._ts.extend(new_ts)
+        self._types.append(t)
+
+    def finalize(self):
+        T = lca_of_types(self._types)
+        return convertbits(self._ts, None, T, checkbit=True)
```

Comment:
I don't think this works with non array of bits?

Reply:
yeah fixed that. good call